aws_lambda_powertools/utilities/advanced_parser/envelopes/base.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+from pydantic import BaseModel, ValidationError
+
+logger = logging.getLogger(__name__)
+
+
+class BaseEnvelope(ABC):
+    def _parse_user_dict_schema(self, user_event: Dict[str, Any], schema: BaseModel) -> Any:
+        if user_event is None:
+            return None
+        logger.debug("parsing user dictionary schema")
+        try:
+            return schema(**user_event)
+        except (ValidationError, TypeError):
+            logger.exception("Validation exception while extracting user custom schema")
+            raise
+
+    def _parse_user_json_string_schema(self, user_event: str, schema: BaseModel) -> Any:
+        if user_event is None:
+            return None
+        logger.debug("parsing user dictionary schema")
+        if schema == str:
+            logger.debug("input is string, returning")
+            return user_event
+        logger.debug("trying to parse as json encoded string")
+        try:
+            return schema.parse_raw(user_event)
```

Comment:
docs: could you add a comment as a context for parse_raw for non-pydantic maintainers?